src/lib.rs

```diff
@@ -319,13 +303,30 @@ pub fn encrypt_and_generate_constraints<F: Field>(
         computed_ciphertext.extend_from_slice(&ciphertext_chunk);
     }
 
+    use ark_r1cs_std::R1CSVar;
+
+    // we insert the computed ciphertext as a public input of the circuit
+    for byte in &computed_ciphertext {
+        let value = byte.value().map_err(|e| anyhow!("{}", e))?;
+        /*UInt8::<ConstraintF>::new_input(
+            constraint_system.clone(),
+            || Ok(value),
+        )?
+
+        */
+        //ciphertext_circuit.push();
+    }
+    /*
+    computed_ciphertext
+
```

Comment:
Instead of "cloning" the last state of `computed_ciphertext` we could tell the gadgets if we want the result to be public or witness. 

We could achieve this by making the gadgets use `new_variable` instead of `new_witness`/`new_input` auditioning the fact that we'd need to pass the `AllocationMode` as a parameter for the gadget.

Reply:
thanks for the suggestion :)

Reply:
This suggestion requires more work than expected so we are not going to go through this path at least for now.